pkg/driver/trace_driver.go

```diff
@@ -100,13 +103,232 @@ func (d *Driver) runCSVWriter(records chan interface{}, filename string, writerD
 	writerDone.Done()
 }
 
-func DAGCreation(functions []*common.Function) *list.List {
-	linkedList := list.New()
-	// Assigning nodes one after another
-	for _, function := range functions {
-		linkedList.PushBack(function)
+func createDAGWorkflow(functionList []*common.Function, function *common.Function, maxWidth int, maxDepth int) *list.List {
+	// Generating the number of Invocations per Depth
+	widthList := []int{}
+	widthList = append(widthList, 1)
+	for i := 1; i < maxDepth-1; i++ {
+		widthList = append(widthList, (rand.Intn(maxWidth-widthList[i-1]+1) + widthList[i-1]))
 	}
-	return linkedList
+	widthList = append(widthList, maxWidth)
```

Comment:
I am deeply grateful for your revisions! The logic of this function is indeed correct. However, I suggest further optional modifications to facilitate future maintenance. For instance:

1. Based on my understanding, lines 108-113 and 161-172 both seem to be generating the DAG structure. May I ask why the entire DAG structure isn't generated first before populating it with functions?

2. I believe the purpose of functionIndex is to prevent function duplication in the DAG. Would it be possible to avoid using functionIndex by first generating the DAG structure and then traversing each layer in a loop? 

3. It appears that some variables could be removed, such as totalFunctionsToInvoke.